client/src/main/scala/drt/client/components/FlightTableContent.scala

```diff
@@ -66,95 +66,111 @@ object FlightTableContent {
         a.viewEnd == b.viewEnd
   }
 
+  case class Model(airportInfos: Map[PortCode, Pot[AirportInfo]]) extends UseValueEq
+
   def apply(shortLabel: Boolean = false,
             originMapper: PortCode => VdomNode = portCode => portCode.toString,
             splitsGraphComponent: SplitsGraphComponentFn = (_: SplitsGraph.Props) => <.div()
            ): Component[Props, Unit, Unit, CtorType.Props] = ScalaComponent.builder[Props]("ArrivalsTableContent")
     .render_PS { (props, _) =>
+      val modelRCP = SPACircuit.connect(m => Model(airportInfos = m.airportInfos))
+
       val flightDisplayFilter = props.airportConfig.portCode match {
         case PortCode("LHR") => LhrFlightDisplayFilter(props.redListUpdates, (portCode, _, _) =>
           props.redListPorts.contains(portCode), LhrTerminalTypes(LhrRedListDatesImpl))
         case _ => DefaultFlightDisplayFilter
       }
 
-      val flightsForWindow = props.portState.window(props.viewStart, props.viewEnd, props.paxFeedSourceOrder)
-      val flights: Seq[ApiFlightWithSplits] = flightsForWindow.flights.values.toList
-        .filter(f => f.apiFlight.flightCodeString.toLowerCase.contains(props.filterFlightNumber.toLowerCase))
-      flights
-        .groupBy(f =>
-          (f.apiFlight.Scheduled, f.apiFlight.Terminal, f.apiFlight.Origin)
-        )
-        .foreach { case (_, flights) =>
-          if (flights.size > 1) {
-            flights.foreach { f =>
-              val flightCode = f.apiFlight.flightCodeString
-              val paxSources = f.apiFlight.PassengerSources.map(ps => s"${ps._1.name}: ${ps._2.actual}").mkString(", ")
-              val splitSources = f.splits.map(s => s"${s.source.toString}: ${s.totalPax}").mkString(", ")
-              log.info(s"Codeshare flight ${SDate(f.apiFlight.Scheduled).prettyDateTime} $flightCode :: $paxSources :: $splitSources")
+      def filterFlights(flights: List[ApiFlightWithSplits],
+                        filter: String,
+                        airportInfos: Map[PortCode, Pot[AirportInfo]]): List[ApiFlightWithSplits] = {
+        flights.filter(f => f.apiFlight.flightCodeString.toLowerCase.contains(filter.toLowerCase)
+          || f.apiFlight.Origin.iata.toLowerCase.contains(filter.toLowerCase)
+          || airportInfos(f.apiFlight.Origin).fold(false)(_.country.toLowerCase.contains(filter.toLowerCase)))
```

Comment:
This will throw an exception if the origin isn't in the `airportInfos` map.
Also, you can replace `.fold(false)` with `.exists`